contrib/cirrus/lib.sh

```diff
@@ -149,37 +158,37 @@ remove_packaged_buildah_files() {
     fi
 }
 
+# Return a list of environment variables that should be passed through
+# to lower levels (tests in containers, or via ssh to rootless).
+# We return the variable names only, not their values. It is up to our
+# caller to reference values.
+passthrough_envars(){
+    warn "Will pass env. vars. matching the following regex:
+    $PASSTHROUGH_ENV_RE"
+    compgen -A variable | \
+        grep -Ev "$SECRET_ENV_RE" | \
+        grep -E  "$PASSTHROUGH_ENV_RE"
+}
+
 in_podman() {
     req_env_vars IN_PODMAN_NAME GOSRC GOPATH SECRET_ENV_RE HOME
     [[ -n "$@" ]] || \
         die "Must specify FQIN and command with arguments to execute"
-    local envargs
-    local envarg
-    local envname
-    local envval
-    local xchars='[:punct:][:cntrl:][:space:]'
-    local envrx='(^CI.*)|(^CIRRUS)|(^GOPATH)|(^GOCACHE)|(^GOSRC)|(^SCRIPT_BASE)|(.*_NAME)|(.*_FQIN)|(^IN_PODMAN_)|(^DISTRO)|(^BUILDAH)|(^STORAGE_)'
```

Comment:
Laboriously comparing the `-e` args between successful-old-buildah-CI and this one, I see that `BUILDAH_ISOLATION=chroot` and `STORAGE_DRIVER=vfs` are not included in the new run. Try adding those to the passthrough.

Reply:
Good find, Thanks!  And lacking anything else to go on, let's give it a try...

Reply:
...added a `fixme!` commit with this, fingers crossed :crossed_fingers:

Reply:
The original (immediately above) just had `^BUILDAH` and `^STORAGE`, not spelled out. Without knowing the full reasoning behind those, I would kind of lean toward keeping things exactly the way they were.

Reply:
Oh yeah I completely agree.  Must have missed them while transposing into the new variables.  Then because I've been over it a billion times, became blinded to that old regex entirely :stuck_out_tongue_winking_eye:

Reply:
This seems to [have done the trick](https://cirrus-ci.com/task/4733485697466368) w/ early failures at least (tests incomplete at the moment).  So apparently the error:

```
[+1318s] # fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/x86_64/APKINDEX.tar.gz
[+1318s] # ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.10/main: temporary error (try again later)
```

is coming from some storage-related failure pathway? :confused: Must be some `apk` fault-handling oddity.  But it would explain why we never found the issue trying to look at networking problems :grinning: Good eye @edsantiago.